setup.php

```diff
@@ -239,3 +244,115 @@ function plugin_fields_checkFiles($force = false) {
       }
    }
 }
+
+function plugin_fields_exportBlockAsYaml($container_id = null) {
+   global $DB;
+
+   $plugin = new Plugin();
+   $yaml_conf = [];
+
+   if (isset($_SESSION['glpiactiveentities'])
+      && $plugin->isInstalled('fields')
+      && $plugin->isActivated('fields')
+      && Session::getLoginUserID()) {
+
+      if ($DB->tableExists(PluginFieldsContainer::getTable())) {
+         $where = [];
+         $where["is_active"] = true;
+         if ($container_id != null) {
+            $where["id"] = $container_id;
+         }
+         $container_obj = new PluginFieldsContainer();
+         $containers    = $container_obj->find($where);
+
+         foreach ($containers as $container) {
+            $itemtypes = (strlen($container['itemtypes']) > 0)
+            ? json_decode($container['itemtypes'], true)
+            : [];
+
+            foreach ($itemtypes as $itemtype) {
+               $fields_obj = new PluginFieldsField();
+               // to get translation
+               $container["itemtype"] = PluginFieldsContainer::getType();
+               $yaml_conf['container'][$container['id']."-".$itemtype]["id"]     = (int) $container['id'];
+               $yaml_conf['container'][$container['id']."-".$itemtype]["name"]   = PluginFieldsLabelTranslation::getLabelFor($container);
+               $yaml_conf['container'][$container['id']."-".$itemtype]["itemtype"] = $itemtype;
+               $yaml_conf['container'][$container['id']."-".$itemtype]["type"] = $container['type'];
+               $yaml_conf['container'][$container['id']."-".$itemtype]["subtype"] = $container['subtype'];
+               $fields = $fields_obj->find(["plugin_fields_containers_id" => $container['id']]);
+               if (count($fields)) {
+                  foreach ($fields as $field) {
+                     $tmp_field= [];
+                     if ($field['is_active'] && !$field['is_readonly']) {
```

Comment:
Maybe this could be directly filtered in the SQL query.

Reply:
done